osu.Game/Screens/Edit/EditorLoader.cs

```diff
@@ -65,6 +65,8 @@ protected override void LoadComplete()
             base.LoadComplete();
 
             // will be restored via lease, see `DisallowExternalBeatmapRulesetChanges`.
+            if (!(Beatmap.Value is DummyWorkingBeatmap))
+                Ruleset.Value = Beatmap.Value.BeatmapInfo.Ruleset;
```

Comment:
I would've updated the ruleset on `Editor` entrance since the entire screen should never operate on a different ruleset than the beatmap's one, but I think we can have it local to test gameplay until that actually proves to be an issue.

Reply:
Again, this is `EditorLoader`. It is pushed *before* the editor. It does not load the player.

Reply:
Right, have confused it with `EditorPlayerLoader`.